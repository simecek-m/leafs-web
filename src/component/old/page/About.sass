@import "colors.sass"

#about-component
  color: $text-primary
  text-align: center

  h2
    color: $text-secondary
    text-transform: uppercase

  .section
    margin: 50px 0
    font-weight: bold
    font-size: 20px

  a
    color: $text-secondary
    text-decoration: none
    transition: 0.3s ease all

  a:hover
    color: $text-primary

  hr
    width: 80%
    max-width: 600px
    border-color: $text-secondary
    margin-bottom: 50px

  .author-link-icon
    margin: 0 5px

  .lottie-link-icon
    margin-right: 5px

