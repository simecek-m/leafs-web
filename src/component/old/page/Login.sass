@import "colors.sass"

.login-methods
  background: $background-secondary
  width: 40%
  max-width: 500px
  margin: 80px auto
  padding: 25px
  border-radius: 55px

  .login-method
    color: $text-secondary
    cursor: pointer
    height: 50px

    span
      &.authority
        display: inline-block
        color: $text-secondary
        text-decoration: none
        font-size: 28px
        font-weight: bold
        line-height: 50px
        vertical-align: middle

    .icon
      display: inline-block
      height: 40px
      width: 40px
      margin: 0 25px
      float: left
      clear: fix
