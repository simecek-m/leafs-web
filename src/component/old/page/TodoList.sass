@import "colors.sass"

.todo-list-component
  color: $text-primary

  .divider
    width: 70%
    min-width: 400px
    max-width: 700px
    margin: 60px auto 20px auto

  .tasks
    margin-left: 45%
    text-align: left
    user-select: none

