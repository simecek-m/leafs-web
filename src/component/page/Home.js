import Topbar from "component/navigation/Topbar";

export function HomePage() {
  return (
    <div>
      <Topbar />
    </div>
  );
}
