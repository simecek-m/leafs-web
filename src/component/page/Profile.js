import React from "react";
import { connect } from "react-redux";
import jsonwebtoken from "jsonwebtoken";
import Back from "component/navigation/Back";
import Button from "component/button/Button";
import { faDoorOpen } from "@fortawesome/free-solid-svg-icons";
import { logout } from "store/actions";
import Title from "component/common/Title";
import { faUserCircle } from "@fortawesome/free-solid-svg-icons";
import { withTranslation } from "react-i18next";
import Settings from "component/menu/Settings";
import "component/page/Profile.sass";

export function Profile({ user, logout, t }) {
  const userInfo = jsonwebtoken.decode(user);
  return (
    <div className="animated fadeIn">
      <Settings />
      <Back />
      <Title text={t("profile.title")} icon={faUserCircle} />
      <div id="profile-info">
        <div className="row">
          <span className="property">{t("profile.avatar")}</span>
          <img
            className="profile-picture"
            src={userInfo.picture}
            alt="profile"
          />
        </div>
        <div className="row">
          <span className="property">{t("profile.firstName")}</span>
          <span className="value">{userInfo.given_name}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.lastName")}</span>
          <span className="value">{userInfo.family_name}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.email")}</span>
          <span className="value">{userInfo.email}</span>
        </div>
        <div className="row">
          <span className="property">{t("profile.authority")}</span>
          <span className="value">{userInfo.iss}</span>
        </div>
        <div className="row single">
          <Button
            icon={faDoorOpen}
            text={t("profile.logout")}
            onClick={() => logout()}
          />
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(
  mapStateToProps,
  { logout }
)(withTranslation()(Profile));
