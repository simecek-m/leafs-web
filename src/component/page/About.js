import React from "react";
import Title from "component/common/Title";
import { faInfoCircle } from "@fortawesome/free-solid-svg-icons";
import { withTranslation } from "react-i18next";
import Back from "component/navigation/Back";
import Settings from "component/menu/Settings";
import Lottie from "react-lottie";
import loadingAnimation from "assets/animation/loading.json";
import errorAnimation from "assets/animation/error.json";
import writeAnimation from "assets/animation/write.json";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faLink } from "@fortawesome/free-solid-svg-icons";
import { faGithub } from "@fortawesome/free-brands-svg-icons";
import "component/page/About.sass";

export function About({ t }) {
  const loadingAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: loadingAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  const errorAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: errorAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  const writeAnimationOptions = {
    loop: true,
    autoplay: true,
    animationData: writeAnimation,
    rendererSettings: {
      preserveAspectRatio: "xMidYMid slice"
    }
  };

  return (
    <div id="about-component" className="animated fadeIn">
      <Settings />
      <Back />
      <Title text={t("about.title")} icon={faInfoCircle} />
      <div className="section">
        <h2>{t("about.author")}</h2>
        <div>Martin Šimeček</div>
        <div className="links">
          <a
            href="https://github.com/simecek-m"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon className="author-link-icon" icon={faGithub} />
          </a>
          <a
            href="https://simecek.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon className="author-link-icon" icon={faLink} />
          </a>
        </div>
      </div>
      <div className="section">
        <h2>{t("about.animations")}</h2>
        <div>
          {t("about.animationDescription", {
            name: "Skeleton UI",
            author: "Surani's Fun World"
          })}
        </div>
        <a
          href="https://lottiefiles.com/4463-hi"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={loadingAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
        <hr />
        <div>
          {t("about.animationDescription", {
            name: "Not Found",
            author: "Oscar Daniel Martínez Núñez"
          })}
        </div>
        <a
          href="https://lottiefiles.com/4339-not-found"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={errorAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
        <hr />
        <div>
          {t("about.animationDescription", {
            name: "Write",
            author: "kinoko"
          })}
        </div>
        <a
          href="https://lottiefiles.com/5729-write"
          rel="noopener noreferrer"
          target="_blank"
        >
          <FontAwesomeIcon className="lottie-link-icon" icon={faLink} />
          Lottie Files
        </a>
        <Lottie
          options={writeAnimationOptions}
          isClickToPauseDisabled={true}
          width={400}
          height={250}
        />
      </div>
    </div>
  );
}

export default withTranslation()(About);
