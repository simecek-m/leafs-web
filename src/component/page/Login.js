import React from "react";
import "component/page/Login.sass";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faGoogle } from "@fortawesome/free-brands-svg-icons";
import { GoogleLogin } from "react-google-login";
import { connect } from "react-redux";
import { login, setLocale } from "store/actions";
import Title from "component/common/Title";
import { faUserShield } from "@fortawesome/free-solid-svg-icons";
import { withTranslation } from "react-i18next";
import jsonwebtoken from "jsonwebtoken";
import { showDangerNotification } from "notification";
import { GOOGLE_API_CLIENT_ID } from "conf";

export class Login extends React.Component {
  loginSuccessCallback(response) {
    const { t, setLocale, login } = this.props;
    const jwt = jsonwebtoken.decode(response.tokenId);
    if (jwt) {
      setLocale(jwt.locale);
      login(response.tokenId);
    } else {
      showDangerNotification(
        t("login.loginFailedTitle"),
        t("login.loginFailedMessage")
      );
    }
  }

  loginFailCallback(error) {
    const { t } = this.props;
    showDangerNotification(
      t("login.loginFailedTitle"),
      t("login.loginFailedMessage")
    );
  }

  render() {
    const { t } = this.props;
    return (
      <div className="animated fadeIn">
        <Title text={t("login.title")} icon={faUserShield} />
        <h2 className="sub-title">{t("login.description")}</h2>
        <div className="login-methods">
          <GoogleLogin
            clientId={GOOGLE_API_CLIENT_ID}
            onSuccess={response => this.loginSuccessCallback(response)}
            onFailure={error => this.loginFailCallback(error)}
            render={renderProps => (
              <div onClick={renderProps.onClick} className="login-method">
                <FontAwesomeIcon className="icon" icon={faGoogle} />
                <span className="authority">Google</span>
              </div>
            )}
          />
        </div>
      </div>
    );
  }
}

export default connect(null, { login, setLocale })(withTranslation()(Login));
