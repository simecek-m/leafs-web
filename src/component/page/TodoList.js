import React from "react";
import api, { setAuthorizationHeader } from "api";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { faList } from "@fortawesome/free-solid-svg-icons";
import Title from "component/common/Title";
import Loading from "component/animation/Loading";
import Back from "component/navigation/Back";
import Task from "component/task/Task";
import AddTask from "component/task/AddTask";
import "component/page/TodoList.sass";

export class TodoList extends React.Component {
  state = {
    todoList: null,
    loading: true,
    error: false
  };

  componentDidMount() {
    const { user, match } = this.props;
    const id = match.params.id;
    api
      .get(`/todos/${id}`, setAuthorizationHeader(user))
      .then(response =>
        this.setState({
          todoList: response.data,
          loading: false
        })
      )
      .catch(() => this.setState({ loading: false, error: true }));
  }

  updateTask(index) {
    const todoList = this.state.todoList;
    const task = todoList.tasks[index];
    todoList.tasks[index].completed = !todoList.tasks[index].completed;
    this.setState({
      todoList
    });

    const { match, user } = this.props;
    const id = match.params.id;
    api
      .put(`/todos/${id}/tasks/${task._id}`, task, setAuthorizationHeader(user))
      .catch(error => {
        this.showNotification(error);
        todoList.tasks[index].completed = !todoList.tasks[index].completed;
        this.setState({
          todoList
        });
      });
  }

  showNotification(error) {
    if (error.response) {
      // TODO: show notification
    }
  }

  deleteTask(taskId, index) {
    const { match, user } = this.props;
    const id = match.params.id;
    api
      .delete(`/todos/${id}/tasks/${taskId}`, setAuthorizationHeader(user))
      .then(() => {
        const todoList = this.state.todoList;
        todoList.tasks.splice(index, 1);
        this.setState({
          todoList
        });
      })
      .catch(error => {
        console.error(error);
        // TODO: show notification
      });
  }

  addTask = text => {
    if (text && text.length > 0) {
      const { match, user } = this.props;
      const id = match.params.id;
      const task = { text, completed: false };
      api
        .post(`/todos/${id}/tasks`, task, setAuthorizationHeader(user))
        .then(response => {
          const todoList = this.state.todoList;
          todoList.tasks.push(response.data);
          this.setState({
            todoList
          });
        })
        .catch(error => {
          console.error(error);
          // TODO: show notification
        });
    }
  };

  render() {
    const tasks =
      this.state.todoList &&
      this.state.todoList.tasks.map((task, index) => (
        <Task
          text={task.text}
          key={index}
          completed={task.completed}
          onClick={() => this.updateTask(index)}
          onDelete={e => {
            e.stopPropagation();
            this.deleteTask(task._id, index);
          }}
        />
      ));
    return (
      <div className="todo-list-component animated fadeIn">
        <Back />
        <Loading loading={this.state.loading} error={this.state.error}>
          <Title
            icon={faList}
            text={this.state.todoList && this.state.todoList.title}
          />
          <div className="tasks tasks-incomplete">
            {tasks &&
              tasks.filter(taskComponent => !taskComponent.props.completed)}
            <AddTask onAdd={this.addTask} />
          </div>
          <hr className="divider" />
          <div className="tasks tasks-completed">
            {tasks &&
              tasks.filter(taskComponent => taskComponent.props.completed)}
          </div>
        </Loading>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default withRouter(connect(mapStateToProps)(TodoList));
