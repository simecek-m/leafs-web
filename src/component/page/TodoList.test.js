import React from "react";
import { TodoList } from "component/page/TodoList";
import { shallow } from "enzyme";
import api from "api";

const MOCK_ROUTER = {
  params: {
    id: 1
  }
};

const INIT_STATE = {
  loading: true,
  error: false,
  todoList: null
};

const TEST_TODO_LIST_ORIGINAL = {
  title: "Work",
  tasks: [
    {
      _id: 1,
      text: "checkout GitHub",
      completed: false
    },
    {
      _id: 2,
      text: "install Yarn",
      completed: true
    },
    {
      _id: 3,
      text: "fix ISS-895 issue",
      completed: false
    }
  ]
};

const apiGetSpy = jest.spyOn(api, "get");
const apiPostSpy = jest.spyOn(api, "post");
const apiPutSpy = jest.spyOn(api, "put");
const apiDeleteSpy = jest.spyOn(api, "delete");

afterEach(() => {
  apiGetSpy.mockReset();
  apiPostSpy.mockReset();
  apiPutSpy.mockReset();
  apiDeleteSpy.mockReset();
});

describe("TodoList component", () => {
  test("should render default TodoList", () => {
    const wrapper = shallow(<TodoList match={MOCK_ROUTER} />);
    expect(wrapper).toMatchSnapshot();
    const state = wrapper.state();
    expect(state).toEqual(INIT_STATE);
  });

  test("should fetch data from API after mounting", async () => {
    const TEST_TODO_LIST = JSON.parse(JSON.stringify(TEST_TODO_LIST_ORIGINAL));
    apiGetSpy.mockResolvedValue({ data: TEST_TODO_LIST });
    const wrapper = shallow(<TodoList match={MOCK_ROUTER} />, {
      disableLifecycleMethods: true
    });
    await wrapper.instance().componentDidMount();
    expect(apiGetSpy).toHaveBeenCalledTimes(1);
    expect(apiGetSpy).toHaveBeenCalledWith(`/todos/${MOCK_ROUTER.params.id}`, {
      headers: { Authorization: "Bearer undefined" }
    });
    const state = wrapper.state();
    expect(state.todoList).toEqual(TEST_TODO_LIST_ORIGINAL);
    expect(state.loading).toBeFalsy();
    expect(state.error).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  test("should change state after error", async () => {
    apiGetSpy.mockRejectedValue({ status: 401 });
    const wrapper = shallow(<TodoList match={MOCK_ROUTER} />, {
      disableLifecycleMethods: true
    });
    await wrapper.instance().componentDidMount();
    expect(apiGetSpy).toHaveBeenCalledTimes(1);
    expect(apiGetSpy).toHaveBeenCalledWith(`/todos/${MOCK_ROUTER.params.id}`, {
      headers: { Authorization: "Bearer undefined" }
    });
    await wrapper.update();
    const state = wrapper.state();
    expect(state.todoList).toBeNull();
    expect(state.loading).toBeFalsy();
    expect(state.error).toBeTruthy();
  });

  test("should add task", async () => {
    const NEW_TASK = { _id: 1, text: "buy milk", completed: false };
    apiPostSpy.mockResolvedValue({ data: NEW_TASK });
    apiGetSpy.mockResolvedValue({ data: { title: "Shop", tasks: [] } });
    const wrapper = await shallow(<TodoList match={MOCK_ROUTER} />);
    let state = wrapper.state();
    expect(apiGetSpy).toHaveBeenCalledTimes(1);
    expect(state.todoList.tasks).toEqual([]);
    await wrapper.instance().addTask(NEW_TASK.text);
    expect(apiPostSpy).toHaveBeenCalledTimes(1);
    expect(apiPostSpy).toHaveBeenCalledWith(
      `/todos/${MOCK_ROUTER.params.id}/tasks`,
      { text: NEW_TASK.text, completed: false },
      { headers: { Authorization: "Bearer undefined" } }
    );
    state = wrapper.state();
    expect(state.todoList.tasks).toEqual([NEW_TASK]);
    expect(wrapper).toMatchSnapshot();
  });

  test("should update task", async () => {
    const UPDATED_TASK_INDEX = 2;
    const TEST_TODO_LIST = JSON.parse(JSON.stringify(TEST_TODO_LIST_ORIGINAL));
    const UPDATED_TODO_LIST = JSON.parse(
      JSON.stringify(TEST_TODO_LIST_ORIGINAL)
    );
    UPDATED_TODO_LIST.tasks[UPDATED_TASK_INDEX].completed = !UPDATED_TODO_LIST
      .tasks[UPDATED_TASK_INDEX].completed;
    apiGetSpy.mockResolvedValue({ data: TEST_TODO_LIST });
    apiPutSpy.mockResolvedValue({});
    const wrapper = await shallow(<TodoList match={MOCK_ROUTER} />);
    const state = wrapper.state();
    expect(state.todoList.tasks).toEqual(TEST_TODO_LIST_ORIGINAL.tasks);
    await wrapper.instance().updateTask(UPDATED_TASK_INDEX);
    expect(apiPutSpy).toHaveBeenCalledTimes(1);
    expect(apiPutSpy).toHaveBeenCalledWith(
      `/todos/${MOCK_ROUTER.params.id}/tasks/${TEST_TODO_LIST.tasks[UPDATED_TASK_INDEX]._id}`,
      UPDATED_TODO_LIST.tasks[UPDATED_TASK_INDEX],
      { headers: { Authorization: "Bearer undefined" } }
    );
    expect(state.todoList.tasks).toEqual(UPDATED_TODO_LIST.tasks);
    expect(wrapper).toMatchSnapshot();
  });

  test("should delete task", async () => {
    const TASK = { _id: 1, text: "buy milk", completed: false };
    const TODO_LIST = { title: "Shop", tasks: [TASK] };
    apiDeleteSpy.mockResolvedValue({});
    apiGetSpy.mockResolvedValue({
      data: JSON.parse(JSON.stringify(TODO_LIST))
    });
    const wrapper = await shallow(<TodoList match={MOCK_ROUTER} />);
    const state = wrapper.state();
    expect(state.error).toBeFalsy();
    expect(state.loading).toBeFalsy();
    expect(state.todoList).toEqual(TODO_LIST);
    await wrapper.instance().deleteTask(TASK._id, 0);
    expect(apiDeleteSpy).toHaveBeenCalledTimes(1);
    expect(apiDeleteSpy).toHaveBeenCalledWith(
      `/todos/${MOCK_ROUTER.params.id}/tasks/${TASK._id}`,
      { headers: { Authorization: "Bearer undefined" } }
    );
    expect(state.todoList.tasks).toEqual([]);
    expect(wrapper).toMatchSnapshot();
  });
});
