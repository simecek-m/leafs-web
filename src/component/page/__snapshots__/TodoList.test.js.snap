// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component should add task 1`] = `
<div
  className="todo-list-component animated fadeIn"
>
  <withRouter(Back) />
  <Loading
    error={false}
    loading={false}
  >
    <Title
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
      text="Shop"
    />
    <div
      className="tasks tasks-incomplete"
    >
      <Task
        completed={false}
        key="0"
        onClick={[Function]}
        onDelete={[Function]}
        text="buy milk"
      />
      <withI18nextTranslation(AddTask)
        onAdd={[Function]}
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="tasks tasks-completed"
    />
  </Loading>
</div>
`;

exports[`TodoList component should delete task 1`] = `
<div
  className="todo-list-component animated fadeIn"
>
  <withRouter(Back) />
  <Loading
    error={false}
    loading={false}
  >
    <Title
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
      text="Shop"
    />
    <div
      className="tasks tasks-incomplete"
    >
      <withI18nextTranslation(AddTask)
        onAdd={[Function]}
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="tasks tasks-completed"
    />
  </Loading>
</div>
`;

exports[`TodoList component should fetch data from API after mounting 1`] = `
<div
  className="todo-list-component animated fadeIn"
>
  <withRouter(Back) />
  <Loading
    error={false}
    loading={false}
  >
    <Title
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
      text="Work"
    />
    <div
      className="tasks tasks-incomplete"
    >
      <Task
        completed={false}
        key="0"
        onClick={[Function]}
        onDelete={[Function]}
        text="checkout GitHub"
      />
      <Task
        completed={false}
        key="2"
        onClick={[Function]}
        onDelete={[Function]}
        text="fix ISS-895 issue"
      />
      <withI18nextTranslation(AddTask)
        onAdd={[Function]}
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="tasks tasks-completed"
    >
      <Task
        completed={true}
        key="1"
        onClick={[Function]}
        onDelete={[Function]}
        text="install Yarn"
      />
    </div>
  </Loading>
</div>
`;

exports[`TodoList component should render default TodoList 1`] = `
<div
  className="todo-list-component animated fadeIn"
>
  <withRouter(Back) />
  <Loading
    error={false}
    loading={true}
  >
    <Title
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
      text={null}
    />
    <div
      className="tasks tasks-incomplete"
    >
      <withI18nextTranslation(AddTask)
        onAdd={[Function]}
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="tasks tasks-completed"
    />
  </Loading>
</div>
`;

exports[`TodoList component should update task 1`] = `
<div
  className="todo-list-component animated fadeIn"
>
  <withRouter(Back) />
  <Loading
    error={false}
    loading={false}
  >
    <Title
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f03a",
            "M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
          ],
          "iconName": "list",
          "prefix": "fas",
        }
      }
      text="Work"
    />
    <div
      className="tasks tasks-incomplete"
    >
      <Task
        completed={false}
        key="0"
        onClick={[Function]}
        onDelete={[Function]}
        text="checkout GitHub"
      />
      <withI18nextTranslation(AddTask)
        onAdd={[Function]}
      />
    </div>
    <hr
      className="divider"
    />
    <div
      className="tasks tasks-completed"
    >
      <Task
        completed={true}
        key="1"
        onClick={[Function]}
        onDelete={[Function]}
        text="install Yarn"
      />
      <Task
        completed={true}
        key="2"
        onClick={[Function]}
        onDelete={[Function]}
        text="fix ISS-895 issue"
      />
    </div>
  </Loading>
</div>
`;
